library/general/src/modules/Linuxrc.rb

```diff
@@ -211,6 +211,31 @@ def SaveInstallInf(root)
       true
     end
 
+    # Returns value of a given Linxurc key/feature defined on commandline
+    # and written into install.inf
+    #
+    # @param [String] key
+    # @return [String] value of a given key
+    def value_for(feature_key)
+      ReadInstallInf()
+
+      feature_key = polish(feature_key)
+      install_inf_features = deep_copy(@install_inf)
+
+      # Some values written on Linuxrc commandline are not known to Linuxrc
+      # Unless they are also mentioned as `PTOptions`, they will appear as "Cmdline" entry
+      install_inf_features.fetch("Cmdline", "").split.map do |cmdline_entry|
+        key_val = cmdline_entry.split("=")
+        install_inf_features[key_val[0]] = key_val[1]
+      end
+
+      matching_keys = install_inf_features.keys.select { |key| polish(key) == feature_key }
+      return nil if matching_keys.empty?
+
+      # The last key wins as in Linuxrc
+      matching_keys.map { |key| install_inf_features[key] }.last
```

Comment:
That's inefficient, you evaluate all matching keys but use just the last value. You can use the last matching key directly:

```ruby
install_inf_features[matching_keys.last]
```